src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -7,7 +7,18 @@ import chisel3._
 import chisel3.internal.sourceinfo.{SourceInfo, SourceInfoTransform}
 
 object BitPat {
-
+  implicit class RecordToBitPat[T <: Record](x: T) {
+    def bp[T <: Record](terms: x.type => Map[Data, BitPat]): BitPat ={
+      x.allElements.map{ e =>
+        terms(x).get(e) match {
+          case Some(value) =>
+            require(value.getWidth == e.getWidth, "BitPat width and data width not equal failed to construct BitPat.")
+            value
+          case None => chisel3.util.BitPat.dontCare(e.getWidth)
+        }
+      }
+    }.reduce(_ ## _)
+  }
```

Comment:
I would rather this just be a method on `object BitPat` rather than an extension method.